packages/opentelemetry-plugin-postgres/test/pg.test.ts

```diff
@@ -0,0 +1,399 @@
+    : 54320,
+};
+
+const DEFAULT_ATTRIBUTES = {
+  [AttributeNames.COMPONENT]: PostgresPlugin.COMPONENT,
+  [AttributeNames.DB_INSTANCE]: CONFIG.database,
+  [AttributeNames.DB_TYPE]: PostgresPlugin.DB_TYPE,
+  [AttributeNames.PEER_HOSTNAME]: CONFIG.host,
+  [AttributeNames.PEER_ADDRESS]: `jdbc:postgresql://${CONFIG.host}:${CONFIG.port}/${CONFIG.database}`,
+  [AttributeNames.PEER_PORT]: CONFIG.port,
+  [AttributeNames.DB_USER]: CONFIG.user,
+};
+
+const runCallbackTest = (
+  span: Span,
+  attributes: Attributes,
+  events: TimedEvent[],
+  spansLength = 1,
+  spansIndex = 0
+) => {
+  const spans = memoryExporter.getFinishedSpans();
+  assert.strictEqual(spans.length, spansLength);
+  const pgSpan = spans[spansIndex];
+  assertionUtils.assertSpan(pgSpan, SpanKind.CLIENT, attributes, events);
+  assertionUtils.assertPropagation(pgSpan, span);
+};
+
+describe('pg@7.x', () => {
+  let client: pg.Client;
+  const tracer = new NodeTracer();
+  const logger = new NoopLogger();
+  const testPostgres = process.env.RUN_POSTGRES_TESTS; // For CI: assumes local postgres db is already available
+  const testPostgresLocally = process.env.RUN_POSTGRES_TESTS_LOCAL; // For local: spins up local postgres db via docker
+  const shouldTest = testPostgres || testPostgresLocally; // Skips these tests if false (default)
+
+  before(async function() {
+    if (!shouldTest) {
+      // this.skip() workaround
+      // https://github.com/mochajs/mocha/issues/2683#issuecomment-375629901
+      this.test!.parent!.pending = true;
+      this.skip();
+    }
+    tracer.addSpanProcessor(new SimpleSpanProcessor(memoryExporter));
+    if (testPostgresLocally) {
+      testUtils.startDocker();
+    }
+
+    client = new pg.Client(CONFIG);
+    try {
+      await client.connect();
+    } catch (e) {
+      throw e;
+    }
+  });
+
+  after(async () => {
+    if (testPostgresLocally) {
+      testUtils.cleanUpDocker();
+    }
+    await client.end();
+  });
+
+  beforeEach(function() {
+    plugin.enable(pg, tracer, logger);
+  });
+
+  afterEach(() => {
+    memoryExporter.reset();
+    plugin.disable();
+  });
+
+  it('should return a plugin', () => {
+    assert.ok(plugin instanceof PostgresPlugin);
+  });
+
+  it('should have correct moduleName', () => {
+    assert.strictEqual(plugin.moduleName, 'pg');
+  });
+
+  it('should maintain pg module error throwing behavior with bad arguments', () => {
+    const assertPgError = (e: Error) => {
+      const src = e.stack!.split('\n').map(line => line.trim())[1];
+      return /node_modules[/\\]pg/.test(src);
+    };
+
+    assert.throws(
+      () => {
+        (client as any).query();
+      },
+      assertPgError,
+      'pg should throw when no args provided'
+    );
+    assert.doesNotThrow(
+      () => (client as any).query({ foo: 'bar' }, undefined, () => null),
+      'pg should not throw when invalid config args are provided'
+    );
+  });
+
+  describe('#client.query(...)', () => {
```

Comment:
Good call. I added some span status tests on each call to `assertSpan`. I also removed my invalid argument statuses since it would be included in the error message anyways